src/main/java/uk/gov/companieshouse/acsp/manage/users/controller/AcspMembershipsController.java

```diff
@@ -50,8 +56,91 @@ public AcspMembershipsController(
 
   @Override
   public ResponseEntity<AcspMembership> addMemberForAcsp(
-      final String requestId, final String acspNumber, final RequestBodyPost requestBody) {
-    return null;
+      final String xRequestId, final String acspNumber, final RequestBodyPost requestBodyPost) {
+    LOG.info(
+        String.format(
+            "Received request for POST `/%s/memberships` with X-Request-Id: %s, user email: %s, and user role: %s",
+            acspNumber, xRequestId, requestBodyPost.getUserId(), requestBodyPost.getUserRole()));
+
+    if (requestBodyPost.getUserId() == null) {
+      LOG.infoContext(xRequestId, "User ID for the target user not provided", null);
+      throw new BadRequestRuntimeException(PLEASE_CHECK_THE_REQUEST_AND_TRY_AGAIN);
+    }
+
+    AcspDataDao acspDataDao;
+    try {
+      acspDataDao = acspDataService.fetchAcspData(acspNumber);
+    } catch (NotFoundRuntimeException exception) {
+      LOG.infoContext(xRequestId, exception.getLocalizedMessage(), null);
+      throw new BadRequestRuntimeException(PLEASE_CHECK_THE_REQUEST_AND_TRY_AGAIN);
+    }
```

Comment:
Isn't the not found better here? Since it indicates the acsp doesnt exist?

Reply:
The LLD states that this should be the bad request.